### PR TITLE
fix no autoloading journal for "filename.xopp" and "filename.xoj" problem

### PR DESCRIPTION
**Pull request by VivekThazhathattil**
This commit does the following:

If one were to enable journal autoloading (Edit -> Preferences -> Load/Save -> Autoloading Journals)  and if the xournalpp file corresponding to `<name>.pdf` were named `<name>.xopp`, `<name>.xoj`,  `<name>.pdf.xopp` or  `<name>.pdf.xoj`, then loading the pdf would autoload the xournalpp file as well. 

Previously it autoloaded only `<name>.pdf.xopp` and `<name>.pdf.xoj`. The default save file name is `<name>.xopp` for the latest build, so it would be good if the xournalpp file with that name autoloaded too.

### CODE REVIEW

**Review by Technius — CHANGES_REQUESTED** [review]
There's too much code duplication here. Can we simplify this by moving the candidate file paths into a vector and then using aloop?

File: src/control/Control.cpp [hunk 1]
```diff
@@ -2079,18 +2079,17 @@ auto Control::loadPdf(const fs::path& filepath, int scrollToPage) -> bool {
...
     if (settings->isAutoloadPdfXoj()) {
-        fs::path f = filepath;
-        Util::clearExtensions(f);
-        f += ".xopp";
-        Document* tmp = loadHandler.loadDocument(f);
-
-        if (tmp == nullptr) {
+        fs::path f;
+        Document* tmp;
+        std::vector<std::string> exts = {".xopp", ".xoj", ".pdf.xopp", ".pdf.xoj"};
```
Comment (Technius) [inline src/control/Control.cpp:2084]:
```suggestion
        const std::vector<std::string> exts = {".xopp", ".xoj", ".pdf.xopp", ".pdf.xoj"};
```

File: src/control/Control.cpp [hunk 2]
```diff
@@ -2079,18 +2079,17 @@ auto Control::loadPdf(const fs::path& filepath, int scrollToPage) -> bool {
     LoadHandler loadHandler;
 
     if (settings->isAutoloadPdfXoj()) {
-        fs::path f = filepath;
-        Util::clearExtensions(f);
-        f += ".xopp";
-        Document* tmp = loadHandler.loadDocument(f);
-
-        if (tmp == nullptr) {
+        fs::path f;
```
Comment (Technius) [inline src/control/Control.cpp:2082]: Move this into the loop since it's only used there.

File: src/control/Control.cpp [hunk 3]
```diff
@@ -2079,18 +2079,17 @@ auto Control::loadPdf(const fs::path& filepath, int scrollToPage) -> bool {
...
+        Document* tmp;
+        std::vector<std::string> exts = {".xopp", ".xoj", ".pdf.xopp", ".pdf.xoj"};
+        for (int i = 0; i < exts.size(); i++) {
```
Comment (Technius) [inline src/control/Control.cpp:2085]: `i` is only used for `exts[i]`, so prefer the range based for loop:
```
for (const std::string& ext: exts) {
```

**Review by Technius — COMMENTED** [review]
Nice work, this is much better. I just have some minor style suggestions (see below).

**Comment by VivekThazhathattil** [comment]
I think it fixes #2911.

**Review by Technius — APPROVED** [review]
Thanks for the find. I've confirmed that this PR fixes the issue and works as intended.

**Pull request merged**